### PR TITLE
Implement get index mapping in ES storage engine

### PR DESCRIPTION
**Pull request by dai-chen**
*Issue #, if available:*

*Description of changes:* Add get index mapping logic to Elasticsearch storage engine. The method in ES storage is used by query engine to fetch index mapping and do analysis, such as semantic check.

*Implementation*:

 1. **ElasticsearchStorageEngine**: Storage engine implementation for Elasticsearch which simply returns *ElasticsearchIndex* for now.
 2. **ElasticsearchIndex**: Implement get index mapping (field types) method by `ElasticsearchClient` underlying.
 3. **ElasticsearchClient**: Abstraction for different Elasticsearch clients. In our plan, apart from a `ElasticsearchNodeClient`, another `ElasticsearchRestClient` will be added soon for entire engine running outside Elasticsearch container.

*Testing*: Added and passed all unit test and coverage required. Integration test will be added later when executor is in place.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: elasticsearch/build.gradle [hunk 1]
```diff
@@ -1,10 +1,65 @@
...
+            limit {
+                minimum = 1.0
+            }
+
+        }
+    }
+    afterEvaluate {
+        classDirectories.setFrom(files(classDirectories.files.collect {
+            fileTree(dir: it,
+                    exclude: ['**/ast/**'])
```
Comment (penghuo) [inline elasticsearch/build.gradle:60]: remove this?

File: elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/mapping/IndexMapping.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+     * @param transform     transform function to transform field type in string to another type
+     * @param <Type>        expected field type class
+     * @return              mapping from field name to field type
+     */
+    public <Type> Map<String, Type> getAllFieldTypes(Function<String, Type> transform) {
+        return fieldMappings.entrySet().
+                             stream().
+                             collect(Collectors.toMap(
+                                 Map.Entry::getKey,
+                                 e -> transform.apply(e.getValue())
```
Comment (penghuo) [inline elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/mapping/IndexMapping.java:78]: The null value should be filtered out.

File: elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/mapping/IndexMapping.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    public int size() {
+        return fieldMappings.size();
+    }
+
+    /**
+     * Return field type by its name.
+     * @param fieldName     field name
+     * @return              field type in string
+     */
+    public String getFieldType(String fieldName) {
```
Comment (penghuo) [inline elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/mapping/IndexMapping.java:63]: Is it possible to return null value? or we accept null value as return value?

File: elasticsearch/build.gradle (hunk 1 above)
Comment (dai-chen) [reply elasticsearch/build.gradle:60]: Sure. Copy-pasted by mistake.

File: elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/mapping/IndexMapping.java (hunk 3 above)
Comment (dai-chen) [reply elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/mapping/IndexMapping.java:63]: Yeah, it is. For now this is used in UT. Will add an existential check method or refactor it to return `Optional` if in use in future.

File: elasticsearch/build.gradle (hunk 1 above)
Comment (dai-chen) [reply elasticsearch/build.gradle:60]: Added JavaDoc and UT. Thanks.

File: elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/mapping/IndexMapping.java (hunk 2 above)
Comment (dai-chen) [reply elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/mapping/IndexMapping.java:78]: Thanks. Double checked there is only null value in Elasticsearch so we should be good. https://www.elastic.co/guide/en/elasticsearch/reference/current/sql-data-types.html

**Pull request merged**